static/js/stream_data.js

```diff
@@ -379,6 +379,21 @@ export function subscribed_stream_ids() {
     return subscribed_subs().map((sub) => sub.stream_id);
 }
 
+export function get_subscribed_streams_of_user(user_id) {
+    const all_subs = get_unsorted_subs();
+    const subscribed_subs = [];
+    for (const sub of all_subs) {
+        // Checks we do inside is_user_subscribed ensure
+        // we return only those streams whose subscribers
+        // current user is allowed to see.
```

Comment:
This comment doesn't make sense to me; we don't need an additional check beyond the `is_user_subscribed` for the target user ID.  The API has already enforced limits on our access to subscriber data correct.

Reply:
Yeah I haven't done any extra checks here apart from `is_user_subscribed`. I've added this just for clarity that we do not need any extra checks as we already do sufficient checks inside `is_user_subscribed`.

Reply:
We do sufficient checks there, but we don't want to log warnings unless we think there's actually a bug in Zulip, so we do need an additional check here.